subprojects/core/src/main/java/org/gradle/testfixtures/internal/ProjectBuilderImpl.java

```diff
@@ -124,6 +125,11 @@ public Project createProject(String name, File inputProjectDir, File gradleUserH
         gradle.setRootProject(project);
         gradle.setDefaultProject(project);
 
+        // Initialize progress logger infrastructure
+        // This is required if Task.execute is called
+        BuildProgressLogger progressLogger = project.getServices().get(BuildProgressLogger.class);
+        progressLogger.buildStarted();
```

Comment:
Shouldn't we be calling all the other methods on `BuildProgressLogger` as well?

Reply:
I don't think so.
This is not about simulating the whole build lifecycle - it is only making sure that `Task.execute` can be called. A `ProjectBuilder` test should have no business with `BuildProgressLogger`.

Moreover, looking at the other methods of `BuildProgressLogger`, I don't think we can call any more since configuring still happens after the root project is created.

WDYT?

Reply:
I'm fine as long as we commit to retiring `ProjectBuilder` at some point.